### DISCUSSION TITLE
Could you please exaplain what structure to use when I need to share one DB model with multiple modules

### DISCUSSION THREAD

**Discussion by beeyev** [nWidart/laravel-modules#1399] [Q&A] [answered]
I will try to explain my thoughts.
Imagine we created a module which implements some "User Profile" functionality, this module has a model `UserPorfile` which works with `user_profiles` database table. It works correctly, no questions here.
AND
There is another module, which also needs to somehow interact with `user_profiles` database table, so, do we need to create another `UserPorfile`  model also in this module? Or we should access model from "User Profile" module? But in this case it breaks the idea of dividing to modules?

Hope I explained my problem clearly.

What is the best strategy here?

**Comment by dcblogdev** [comment, marked as answer]
I understand what you mean. I use imports from other modules it does create a dependencies with those modules but I think that's fine. Often modules cannot be fully independent.
